src/main/java/com/arpnetworking/tsdcore/sinks/DimensionFilteringSink.java

```diff
@@ -33,6 +34,12 @@
      */
     @Override
     public void recordAggregateData(final PeriodicData data) {
+        if (_whitelistDimensions.isPresent()) {
+            if (!_whitelistDimensions.get().containsAll(data.getDimensions().keySet())) {
```

Comment:
I don't think "white-list" is the appropriate term here. This is the set of required dimensions.